docs/content/integrations/tableau.mdx

```diff
@@ -151,3 +151,62 @@ defs = Definitions.merge(
     marketing_team_workspace.build_defs(),
 )
 ```
+
+### Add a Data Quality Warning in Tableau using a sensor
+
+When an upstream dependency of a Tableau asset fails to materialize or to pass the asset checks, it is possible to add a [Data Quality Warning](https://help.tableau.com/current/online/en-us/dm_dqw.htm) to the corresponding data source in Tableau. This can be achieved by leveraging the `add_data_quality_warning_to_data_source` in a sensor.
+
+```python file=/integrations/tableau/add-tableau-data-quality-warning.py
+from dagster_tableau import TableauCloudWorkspace
+
+from dagster import (
+    Definitions,
+    EnvVar,
+    RunFailureSensorContext,
+    asset,
+    run_failure_sensor,
+)
+
+# Connect to Tableau Cloud using the connected app credentials
+workspace = TableauCloudWorkspace(
+    connected_app_client_id=EnvVar("TABLEAU_CONNECTED_APP_CLIENT_ID"),
+    connected_app_secret_id=EnvVar("TABLEAU_CONNECTED_APP_SECRET_ID"),
+    connected_app_secret_value=EnvVar("TABLEAU_CONNECTED_APP_SECRET_VALUE"),
+    username=EnvVar("TABLEAU_USERNAME"),
+    site_name=EnvVar("TABLEAU_SITE_NAME"),
+    pod_name=EnvVar("TABLEAU_POD_NAME"),
+)
+
+
+@asset(
+    metadata={"dagster/tableau_data_source_id": "f5660c7-2b05-4ff0-90ce-3199226956c6"}
```

Comment:
Not really sure I get what is happening here, what's the meaning of this metadata tag? It wasn't explained earlier

Reply:
I added in comment in [e3213d2](https://github.com/dagster-io/dagster/pull/25078/commits/e3213d2d205af5578260a9900ffff9c26010ac0c).

Currently, we must know the ID of the Tableau data source to which an upstream resource corresponds in order to use the API endpoint. This should be happening behind the scene at some point.